src/github_sdk.py

```diff
@@ -1,101 +1,135 @@
+            "data": {
+                "job": job["html_url"],
+            },
+            "tags": {
+                "job_status": job["conclusion"],  # e.g. success, failure, skipped
+                "branch": runs["head_branch"],
+                "commit": runs["head_sha"],
+                "repo": repo,
+                "run_attempt": runs["run_attempt"],  # Rerunning a job
+                # It allows querying jobs within the same workflow (e.g. foo.yml)
+                "workflow": workflow["path"].rsplit("/")[-1],
+            },
+        }
+        if runs.get("pull_requests"):
+            pr_number = runs["pull_requests"][0]["number"]
+            meta["data"]["pr"] = f"https://github.com/{repo}/pull/{pr_number}"
+            meta["tags"]["pull_request"] = pr_number
+
+        return meta
+
+    # Documentation about traces, transactions and spans
+    # https://docs.sentry.io/product/sentry-basics/tracing/distributed-tracing/#traces
+    # https://develop.sentry.dev/sdk/performance/
+    def _generate_trace(self, job):
+        meta = self._get_extra_metadata(job)
+        transaction = _base_transaction(job)
+        transaction["user"] = meta["author"]
+        transaction["tags"] = meta["tags"]
+        transaction["contexts"]["trace"]["data"] = meta["data"]
+
+        # Transactions have name, spans don't.
+        transaction["contexts"]["trace"]["op"] = job["name"]
+        transaction["contexts"]["trace"]["description"] = job["name"]
+        transaction["contexts"]["trace"][
+            "status"
+        ] = self.github_status_trace_status.get(job["conclusion"], "unimplemented")
+        transaction["spans"] = _generate_spans(
+            job["steps"],
+            transaction["contexts"]["trace"]["span_id"],
+            transaction["contexts"]["trace"]["trace_id"],
+        )
+        return transaction
+
+    def _send_envelope(self, trace):
+        envelope = Envelope()
+        envelope.add_transaction(trace)
+        now = datetime.utcnow()
+
+        headers = {
+            "event_id": get_uuid(),  # Does this have to match anything?
+            "sent_at": format_timestamp(now),
+            "Content-Type": "application/x-sentry-envelope",
+            "Content-Encoding": "gzip",
+            "X-Sentry-Auth": f"Sentry sentry_key={self.sentry_key},"
+            + f"sentry_client=gha-sentry/0.0.1,sentry_timestamp={now},"
+            + "sentry_version=7",
+        }
+
+        body = io.BytesIO()
+        with gzip.GzipFile(fileobj=body, mode="w") as f:
+            envelope.serialize_into(f)
+
+        req = requests.post(
+            self.sentry_project_url, data=body.getvalue(), headers=headers
+        )
+        req.raise_for_status()
+        return req
+
+    def send_trace(self, job):
+        # This can happen when the workflow is skipped and there are no steps
+        if job["conclusion"] == "skipped":
```

Comment:
In the future I will be able to configure ignoring events that are not cause by a push or pull_request (e.g. an issue label is changed).